diff --git a/pym/repoman/modules/scan/ebuild/checks.py b/pym/repoman/modules/scan/ebuild/checks.py
index 15e225156..c0dd8fcb4 100644
--- a/pym/repoman/modules/scan/ebuild/checks.py
+++ b/pym/repoman/modules/scan/ebuild/checks.py
@@ -86,6 +86,7 @@ class EbuildHeader(LineCheck):
 	repoman_check_name = 'ebuild.badheader'
 
 	gentoo_copyright = r'^# Copyright ((1999|2\d\d\d)-)?%s Gentoo Foundation$'
+	os_copyright = r'^# Copyright ((1999|2\d\d\d)-)?%s Obsidian-Studios, Inc.$'
 	gentoo_license = (
 		'# Distributed under the terms'
 		' of the GNU General Public License v2')
@@ -99,12 +100,15 @@ class EbuildHeader(LineCheck):
 			self.modification_year = str(time.gmtime(pkg.mtime)[0])
 		self.gentoo_copyright_re = re.compile(
 			self.gentoo_copyright % self.modification_year)
+		self.os_copyright_re = re.compile(
+			self.os_copyright % self.modification_year)
 
 	def check(self, num, line):
 		if num > 2:
 			return
 		elif num == 0:
-			if not self.gentoo_copyright_re.match(line):
+			if not ( self.gentoo_copyright_re.match(line) or
+			  self.os_copyright_re.match(line) ):
 				return errors.COPYRIGHT_ERROR
 		elif num == 1 and line.rstrip('\n') != self.gentoo_license:
 			return errors.LICENSE_ERROR
