diff --git a/src/main/java/facebook4j/Page.java b/src/main/java/facebook4j/Page.java
index abcf10b..7988ac7 100644
--- a/src/main/java/facebook4j/Page.java
+++ b/src/main/java/facebook4j/Page.java
@@ -18,6 +18,7 @@ package facebook4j;
 
 import java.net.URL;
 import java.util.Date;
+import java.util.Map;
 
 /**
  * @author Ryuji Yamashita - roundrop at gmail.com
@@ -62,4 +63,6 @@ public interface Page {
     String getAbout();
     String getUsername();
     String getMission();
+    Map<String,String> getHours();
+
 }
diff --git a/src/main/java/facebook4j/internal/json/PageJSONImpl.java b/src/main/java/facebook4j/internal/json/PageJSONImpl.java
index 2d9f316..93d444c 100644
--- a/src/main/java/facebook4j/internal/json/PageJSONImpl.java
+++ b/src/main/java/facebook4j/internal/json/PageJSONImpl.java
@@ -25,6 +25,7 @@ import facebook4j.internal.org.json.JSONObject;
 
 import java.net.URL;
 import java.util.Date;
+import java.util.Map;
 
 import static facebook4j.internal.util.z_F4JInternalParseUtil.*;
 
@@ -59,6 +60,7 @@ import static facebook4j.internal.util.z_F4JInternalParseUtil.*;
     private String about;
     private String username;
     private String mission;
+    private Map<String,String> hours;
 
     /*package*/PageJSONImpl(HttpResponse res, Configuration conf) throws FacebookException {
         super(res);
@@ -109,6 +111,7 @@ import static facebook4j.internal.util.z_F4JInternalParseUtil.*;
             about = getRawString("about", json);
             username = getRawString("username", json);
             mission = getRawString("mission", json);
+            hours = getStringMap("hours", json);
 
         } catch (JSONException jsone) {
             throw new FacebookException(jsone.getMessage(), jsone);
@@ -211,6 +214,10 @@ import static facebook4j.internal.util.z_F4JInternalParseUtil.*;
         return mission;
     }
 
+    public Map<String,String> getHours() {
+        return hours;
+    }
+
     /*package*/
     static ResponseList<Page> createPageList(HttpResponse res, Configuration conf) throws FacebookException {
         try {
@@ -263,7 +270,8 @@ import static facebook4j.internal.util.z_F4JInternalParseUtil.*;
                 + ", fanCount=" + fanCount
                 + ", id=" + id + ", name=" + name + ", category=" + category
                 + ", createdTime=" + createdTime + ", about=" + about
-                + ", username=" + username + ", mission=" + mission + "]";
+                + ", username=" + username + ", mission=" + mission
+                + ", hours=" + hours +"]";
     }
 
 }
