diff --git a/src/main/java/org/springframework/scripting/jruby/JRubyScriptFactory.java b/src/main/java/org/springframework/scripting/jruby/JRubyScriptFactory.java
index a7fb53d427..481a404721 100644
--- a/src/main/java/org/springframework/scripting/jruby/JRubyScriptFactory.java
+++ b/src/main/java/org/springframework/scripting/jruby/JRubyScriptFactory.java
@@ -111,8 +111,8 @@ public class JRubyScriptFactory implements ScriptFactory, BeanClassLoaderAware {
 		}
 		catch (RaiseException ex) {
 			RubyException rubyEx = ex.getException();
-			String msg = (rubyEx != null && rubyEx.message != null) ?
-					rubyEx.message.toString() : "Unexpected JRuby error";
+			String msg = (rubyEx != null) ?
+					rubyEx.getMessage().toString() : "Unexpected JRuby error";
 			throw new ScriptCompilationException(scriptSource, msg, ex);
 		}
 		catch (JumpException ex) {
