diff -Naur a/src/main/java/org/apache/logging/log4j/message/MapMessage.java b/src/main/java/org/apache/logging/log4j/message/MapMessage.java
--- a/src/main/java/org/apache/logging/log4j/message/MapMessage.java	2017-02-05 21:04:09.937828866 -0500
+++ b/src/main/java/org/apache/logging/log4j/message/MapMessage.java	2017-02-05 21:05:36.713875560 -0500
@@ -246,7 +246,7 @@
     public void asXml(final StringBuilder sb) {
         sb.append("<Map>\n");
         for (int i = 0; i < data.size(); i++) {
-            sb.append("  <Entry key=\"").append(data.getKeyAt(i)).append("\">").append(data.getValueAt(i))
+            sb.append("  <Entry key=\"").append(data.getKeyAt(i)).append("\">").append((String)data.getValueAt(i))
                     .append("</Entry>\n");
         }
         sb.append("</Map>");
