diff -Naur a/src/main/java/org/apache/logging/log4j/core/pattern/MapPatternConverter.java b/src/main/java/org/apache/logging/log4j/core/pattern/MapPatternConverter.java
--- a/src/main/java/org/apache/logging/log4j/core/pattern/MapPatternConverter.java	2017-02-05 21:27:38.038586575 -0500
+++ b/src/main/java/org/apache/logging/log4j/core/pattern/MapPatternConverter.java	2017-02-05 21:28:00.678598759 -0500
@@ -79,7 +79,7 @@
                 if (i > 0) {
                     toAppendTo.append(", ");
                 }
-                toAppendTo.append(sortedMap.getKeyAt(i)).append('=').append(sortedMap.getValueAt(i));
+                toAppendTo.append(sortedMap.getKeyAt(i)).append('=').append((String)sortedMap.getValueAt(i));
             }
             toAppendTo.append('}');
         } else {
