diff --git a/src/aQute/lib/collections/DoubleKeyMap.java b/src/aQute/lib/collections/DoubleKeyMap.java
index 4107f31..40ecfdf 100644
--- a/src/aQute/lib/collections/DoubleKeyMap.java
+++ b/src/aQute/lib/collections/DoubleKeyMap.java
@@ -97,20 +97,6 @@ public class DoubleKeyMap<K1, K2, V> extends HashMap<K1,Map<K2,V>>implements Map
 		return result;
 	}
 
-	public V remove(K1 key1, K2 key2) {
-		assert k1Class.isInstance(key1);
-		assert k2Class.isInstance(key2);
-
-		Map<K2,V> set = get(key1);
-		if (set == null) {
-			return null;
-		}
-		V result = set.remove(key2);
-		if (set.isEmpty())
-			remove(key1);
-		return result;
-	}
-
 	public Iterator<Map.Entry<K2,V>> iterate(K1 key) {
 		assert k1Class.isInstance(key);
 		Map<K2,V> set = get(key);
